rules/no-zero-fractions.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const message = 'Zero fraction or dangling dot in number.';
+
+const regexpEndsWithZeroFraction = /\.0*$/;
+const create = context => {
+	return {
+		Literal: node => {
+			if (typeof node.value === 'number' && node.value % 1 === 0 && regexpEndsWithZeroFraction.test(node.raw)) {
```

Comment:
Will this also catch `1.10`?

Reply:
This implementation is specifically limited to only work with integer numbers. It also doesn't react to numbers in scientific notation.
I just did it the way it was said in original issue, but if my interpretation isn't correct, I can add both other cases. Honestly speaking, I can't make a judgement about extent of consistency we want here(mostly because while `1.0` and `1` are equal in JS, they actually have different behavior, while `1.10` and `1.1` are really always the same as far as I know).

Reply:
> Will this also catch 1.10?

Yes, it should handle this too. Sorry, missed that in the original issue.

Reply:
Added that, also added scientific notation `1.10e123` => `1.1e123`.